js/apps/admin-ui/src/clients/authorization/ResourcesPolicySelect.tsx

```diff
@@ -108,57 +128,141 @@ export const ResourcesPolicySelect = ({
     [search],
   );
 
+  const [toggleUnsavedChangesDialog, UnsavedChangesConfirm] = useConfirmDialog({
+    titleKey: t("unsavedChangesTitle"),
+    messageKey: t("unsavedChangesConfirm"),
+    continueButtonLabel: t("common:continue"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: () => {
+      navigate(
+        toPolicyDetails({
+          realm: realm,
+          id: clientId,
+          policyId: clickedPolicy?.id!,
+          policyType: clickedPolicy?.type!,
+        })
+      );
+    },
+  });
+
   const toSelectOptions = () =>
     items.map((p) => (
       <SelectOption key={p.id} value={p.id}>
         {p.name}
       </SelectOption>
     ));
 
+  const toChipGroupItems = (
+    field: ControllerRenderProps<PolicyRepresentation, Type>
+  ) => {
+    return (
+      <ChipGroup>
+        {field.value?.map((permissionId) => {
+          const policy = items.find(
+            (permission) => permission.id === permissionId
+          );
+
+          if (!policy) {
+            return null;
+          }
+
+          const to = toPolicyDetails({
+            realm: realm,
+            id: clientId,
+            policyId: policy.id!,
+            policyType: policy.type!,
+          });
+
+          return (
+            <Chip
+              key={policy.id}
+              onClick={(event) => {
+                event.stopPropagation();
+                field.onChange(field.value?.filter((id) => id !== policy.id));
+              }}
+            >
+              {policy.type ? (
+                <Link
+                  to={to}
+                  onClick={(event) => {
+                    if (isDirty) {
+                      event.preventDefault();
+                      event.stopPropagation();
+                      setOpen(false);
+                      setClickedPolicy(policy);
+                      toggleUnsavedChangesDialog();
+                    } else {
```

Comment:
There should be no need for this `else` clause, as the default behavior should be that the link is followed.